vector/src/main/java/im/vector/app/features/home/room/detail/TimelineFragment.kt

```diff
@@ -2374,6 +2385,29 @@ class TimelineFragment @Inject constructor(
         }
     }
 
+    /**
+     * Display a dialog that will let the user to enable threads
+     */
+
+    private fun displayThreadsBetaNotice() =
+            activity?.let {
+                MaterialAlertDialogBuilder(it)
+                        .setTitle(R.string.threads_beta_enable_notice_title)
+                        .setMessage(threadsManager.getBetaEnableThreadsMessage())
+                        .setCancelable(true)
+                        .setNegativeButton(R.string.action_not_now) { _, _ -> }
+                        .setPositiveButton(R.string.action_try_it_out) { _, _ ->
+                            timelineViewModel.onCleared() // We should first clear our viewModel
```

Comment:
do we need to clear the viewmodel if we're restarting the app? I'm wondering if we'll get it for free~

Reply:
If we don't there is a crash in the live events observers

Reply:
This is only happening if we RR the app from the Timeline, while the observers are active

Reply:
ah that is strange, I expected restarting the app to also call `timelineViewModel.onCleared()`, perhaps there's a existing issue in the screen with the way the listeners are being set/released (if they're tied to the viewmodel lifecycle rather than the fragment)

not a blocker for this PR, just something to bear in mind for future

Reply:
Yes, definitely its something wrong here, manually clearing the viewModel it's a workaround

Reply:
I agree this is quite strange to call `onCleared` manually (there is no other case like this in the codebase). There is probably another bug for your unexpected behavior.